aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/featuretest/generators/testcasegenerators/SignInTestCaseGenerator.kt

```diff
@@ -295,5 +298,32 @@ object SignInTestCaseGenerator : SerializableProvider {
         )
     )
 
-    override val serializables: List<Any> = listOf(baseCase, challengeCase, deviceSRPTestCase, customAuthCase)
+    private val customAuthWithSRPCase = FeatureTestCase(
+        description = "Test that Custom Auth signIn invokes proper cognito request and returns password challenge",
+        preConditions = PreConditions(
+            "authconfiguration.json",
+            "SignedOut_Configured.json",
+            mockedResponses = listOf(
+                mockedInitiateAuthResponse,
+                mockedRespondToAuthCustomChallengeResponse
+            )
+        ),
+        api = API(
+            AuthAPI.signIn,
+            params = mapOf(
+                "username" to username,
+                "password" to "",
+            ).toJsonElement(),
+            options = mapOf(
+                "signInOptions" to
+                        mapOf("authFlow" to AuthFlowType.CUSTOM_AUTH_WITH_SRP.toString())
+            ).toJsonElement()
```

Comment:
Recommended to use `mapOf` as the last resort. Instead, use the actual Objects (Options object) for this instance. This way it is fault tolerant and easy when refactoring for example.